ipn/ipnlocal/cert.go

```diff
@@ -691,13 +692,24 @@ func validCertPEM(domain string, keyPEM, certPEM []byte, roots *x509.CertPool, n
 	if leaf == nil {
 		return false
 	}
-	_, err = leaf.Verify(x509.VerifyOptions{
-		DNSName:       domain,
-		CurrentTime:   now,
-		Roots:         roots,
-		Intermediates: intermediates,
-	})
-	return err == nil
+	for {
+		_, err = leaf.Verify(x509.VerifyOptions{
+			DNSName:       domain,
+			CurrentTime:   now,
+			Roots:         roots,
+			Intermediates: intermediates,
+		})
+		if err == nil {
+			return true
+		}
+		if roots != nil {
+			return false
+		}
+		// If validation failed and they specified nil for roots (meaning to use
+		// the system roots), then give it another chance to validate using the
+		// binary's baked-in roots (LetsEncrypt). See tailscale/tailscale#14690.
+		roots = bakedroots.Get()
+	}
```

Comment:
nit: this infinite loop form is harder to reason about, how about something more explicit:
```suggestion
	for _, roots := []*x509.CertPool{roots, bakedroots.Get()} {
		_, err = leaf.Verify(x509.VerifyOptions{
			DNSName:       domain,
			CurrentTime:   now,
			Roots:         roots,
			Intermediates: intermediates,
		})
		if err == nil {
			return true
		}
	}
	return false
```